tests/main/hidden-snap-dir/task.yaml

```diff
@@ -8,13 +8,79 @@ environment:
     NAME: test-snapd-tools
 
 prepare: |
-    snap pack "$TESTSLIB"/snaps/"$NAME"
+    snap pack "$TESTSLIB/snaps/$NAME"
     "$TESTSTOOLS"/snaps-state install-local "$NAME"
 
 restore: |
     snap unset system experimental.hidden-snap-folder
 
 execute: |
+    # core22 snap isn't available for x86
+    if os.query is-pc-i386; then
+      exit 0
+    fi
+
+    # Checks that the env vars are as expected after the migration. 1st parameter
+    # is the current revision; 2nd parameter toggles ~/Snap related checks
+    # ("true" does the checks, anything else doesn't)
```

Comment:
I would find it more readable without the documentation of the parameters, and having instead assignments just at the beginning of the functions:
```
local REV="$1"
local CHECK_VARS="$2"
```
Especially if the function is longer than a handful lines, because in that case when you see `$1` at the end of the function you might need to scroll up to see what that parameter is meant to carry.

Reply:
That's a good suggestion, it would be more readable and concise. I'll change it, thank you